### PR TITLE
First version of PHOS QC

### PR DESCRIPTION
**Pull request by peressounko**

### CODE REVIEW

**Comment by peressounko** [comment]
Dear @Barthelemy, some checks freezed - this is our first commit it is normal?
   Best regards, Dmitri

**Comment by Barthelemy** [comment]
Dear @peressounko , no it is not normal, @ktf is on it.

**Comment by Barthelemy** [comment]
A couple of things already : There are some lines modified in the framework itself that should probably be reverted. There are also a couple of commits that are not yours (from Adam) that should be removed. One way is to use `git rebase -i HEAD~5` (the number is the number of commits to edit). Also using `git pull --rebase` might avoid pulling these extra commits in the future.

**Comment by knopers8** [comment]
For first-time contributors the builds aren't run anyway until the PR gets a positive review

**Comment by peressounko** [comment]
Dear @Barthelemy, I destroyed O2 between commits, had to pull and got external commits. Now I removed them. Best regards, Dmitri

**Comment by knopers8** [comment]
Could you please revert the changes outside of the PHOS module, especially in the Framework directory?
Also, are the changes in the EMCal configuration files intended and approved by the EMCal QC team?

**Comment by peressounko** [comment]
Changes beyond PHOS were accidental and now removed.

**Comment by peressounko** [comment]
Dear @Barthelemy , @knopers8 I updated commit with PHOS-only stuff. Could you please comment?
   Thanks, Dmitri

**Comment by Barthelemy** [comment]
Dear Dmitri, 
I will try to review it today in more details.
Cheers,

File: Modules/PHOS/include/PHOS/DigitCheck.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+#define QC_MODULE_PHOS_PHOSDIGITCHECK_H
+
+#include "QualityControl/CheckInterface.h"
+
+namespace o2::quality_control_modules::phos
+{
+
+/// \brief  Check whether a plot is empty or not.
+///
+/// \author Barthelemy von Haller
```
Comment (Barthelemy) [inline Modules/PHOS/include/PHOS/DigitCheck.h:26]: Please update class description.

File: Modules/PHOS/src/DigitsQcTask.cxx [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+    } else {
+      mTimeE[mod]->Reset();
+    }
+
+    if (!mCellN[mod]) {
+      mCellN[mod] = new TH1F(Form("CellMeanNM%d", mod), Form("Average number of cells in module %d", mod), 1000, 0., 1000.);
+      mCellN[mod]->GetXaxis()->SetTitle("N_{cell}/event");
+      mCellN[mod]->GetYaxis()->SetTitle("dN_{events}/dN_{cell}");
+      try {
+        getObjectsManager()->startPublishing(mCellN[mod]);
```
Comment (Barthelemy) [inline Modules/PHOS/src/DigitsQcTask.cxx:252]: I am not sure to follow the logic here. If the publication of `mCellN[mod]` fails, then we stop publishing the Mean but try again to publish `mCellN[mod]` ?

Comment (peressounko) [reply Modules/PHOS/src/DigitsQcTask.cxx:252]: What I see in code of Manager::publish(), it checks if this object already published and sends exception if positive. Then I stop publishing of old object and start publishing of new one. I do not see other way to check if object is published: it anyway goes through exceptions.

File: Modules/PHOS/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+
+install (
+  DIRECTORY etc DESTINATION Modules/PHOS
+)
+
+# ---- Executables ----
+
+add_executable(qcRunPhosDigit src/runQCPHOSDigits.cxx)
```
[the remark this answers is not in the document]
Comment (peressounko) [reply Modules/PHOS/CMakeLists.txt]: fixed

File: Modules/PHOS/include/PHOS/DigitCheck.h (hunk 1 above)
Comment (peressounko) [reply Modules/PHOS/include/PHOS/DigitCheck.h]: fixed

File: Modules/PHOS/src/DigitsQcTask.cxx (hunk 2 above)
Comment (Barthelemy) [reply Modules/PHOS/src/DigitsQcTask.cxx:252]: Hi, Thanks for this very prompt reply. I misunderstood indeed, this is correct. I am adding a method "isBeingPublished" that should help.

**Comment by Barthelemy** [comment]
waiting for the CI and then we can proceed.

**Pull request merged**